api_v3/lib/KalturaLiveEntryService.php

```diff
@@ -357,40 +387,42 @@ private function createRecordedEntry(LiveEntry $dbEntry, $mediaServerIndex)
 	 * @action unregisterMediaServer
 	 * @param string $entryId Live entry id
 	 * @param string $hostname Media server host name
-	 * @param KalturaMediaServerIndex $mediaServerIndex Media server index primary / secondary
+	 * @param KalturaEntryServerNodeType $mediaServerIndex Media server index primary / secondary
 	 * @return KalturaLiveEntry The updated live entry
 	 *
 	 * @throws KalturaErrors::ENTRY_ID_NOT_FOUND
-	 * @throws KalturaErrors::MEDIA_SERVER_NOT_FOUND
+	 * @throws KalturaErrors::SERVER_NODE_NOT_FOUND
+	 * @throws KalturaErrors::ENTRY_SERVER_NODE_MULTI_RESULT
 	 */
 	function unregisterMediaServerAction($entryId, $hostname, $mediaServerIndex)
 	{
-		$this->dumpApiRequest($entryId);
 		KalturaLog::debug("Entry [$entryId] from mediaServerIndex [$mediaServerIndex] with hostname [$hostname]");
-
-		$dbEntry = entryPeer::retrieveByPK($entryId);
-		if (!$dbEntry || !($dbEntry instanceof LiveEntry))
+		/* @var $dbLiveEntry LiveEntry */
+		$dbLiveEntry = entryPeer::retrieveByPK($entryId);
+		if (!$dbLiveEntry || !($dbLiveEntry instanceof LiveEntry))
 			throw new KalturaAPIException(KalturaErrors::ENTRY_ID_NOT_FOUND, $entryId);
 
-		$dbEntry->unsetMediaServer($mediaServerIndex, $hostname);
+		$dbLiveEntry->unsetMediaServer($mediaServerIndex);
 
-		if(!$dbEntry->hasMediaServer() && $dbEntry->getRecordedEntryId())
-		{
-			$dbEntry->setRedirectEntryId($dbEntry->getRecordedEntryId());
-		}
+		/* @var $dbLiveEntryServerNode LiveEntryServerNode */
+		$dbLiveEntrySeverNode = EntryServerNodePeer::retrieveByEntryIdAndServerType($entryId, $mediaServerIndex);
 
-		if ( count( $dbEntry->getMediaServers() ) == 0 )
+		if ($dbLiveEntrySeverNode)
 		{
-			if ( $dbEntry->getCurrentBroadcastStartTime() )
+			EntryServerNodePeer::doDelete($dbLiveEntrySeverNode);
+			$dbLiveEntry->setLastBroadcastEndTime(kApiCache::getTime());
```

Comment:
Where did this call came from?
previously we did $dbEntry->setCurrentBroadcastStartTime( 0 ); where did ti go to?